final_task/rss_parser/Classes/RSSParser.py

```diff
@@ -0,0 +1,89 @@
+import feedparser
+from Output_functions import getting_full_info, getting_pack_of_news, converting_to_json, \
+    writing_to_file, getting_from_json_to_pack
+import logging
+
+
+class RSSParser:
+    """
+    class RSSParser has 3 parameters and it calls function parse when created
+    """
+
+    def __init__(self, param_url, num_of_news=None, list_of_args=None):
+        self.feed_url = param_url
+        self.number = num_of_news
+        self.list_of_args = list_of_args
+
+    def parse(self):
+        """
+        1. Use feedparser to get the page
+        2. If we have some problems with connection - raise ConnectionError
+        3. Handle Exception without showing a traceback
+        :return:
+        """
+        try:
+            logging.info("Trying to get page from feedparser!")
+            the_feed = feedparser.parse(self.feed_url)
+            logging.info("Got it (the page)!")
+            if the_feed.get('bozo') == 1:
+                if '--date' in self.list_of_args:
+                    self.news_for_date()
+                else:
+                    logging.info("Got some problems due to connection!")
+                    raise ConnectionError
+        except ConnectionError:
+            logging.critical("CONNECTION ERROR, HELP!")
+            print("You have some connection problems!")
+            if '--date' in self.list_of_args:
```

Comment:
You have done the same check in `try` block. Please, refactor this logic

Reply:
Fixed, thanks.